src/java/org/apache/nutch/plugin/URLStreamHandlerFactory.java

```diff
@@ -0,0 +1,85 @@
+package org.apache.nutch.plugin;
+
+import java.lang.ref.WeakReference;
+import java.net.URL;
+import java.net.URLStreamHandler;
+import java.util.ArrayList;
+
+import org.mortbay.log.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class URLStreamHandlerFactory
+    implements java.net.URLStreamHandlerFactory {
+  
+  protected static final Logger LOG = LoggerFactory
+      .getLogger(URLStreamHandlerFactory.class);
+  
+  /** The singleton instance. */
+  private static URLStreamHandlerFactory instance;
+  
+  /** Here we register all PluginRepositories. */
+  private ArrayList<WeakReference<PluginRepository>> prs;
+  
+  static {
+    instance = new URLStreamHandlerFactory();
+    URL.setURLStreamHandlerFactory(instance);
+    LOG.info("Registered URLStreamHandlerFactory with the JVM.");
+  }
+  
+  private URLStreamHandlerFactory() {
+    LOG.debug("URLStreamHandlerFactory()");
+    prs = new ArrayList<>();
+  }
+
+  /** Return the singleton instance of this class. */
+  public static URLStreamHandlerFactory getInstance() {
+    LOG.debug("getInstance()");
```

Comment:
Was this for your local debugging only? Not sure if it should be included in production source.

Reply:
That is why it uses the debug level logging. If such logging should not go to production, you could as well question all debug messages.